extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -275,75 +304,93 @@ export class Bind {
   }
 
   /**
-   * Scans `node` and its descendants and adds bindings for nodes
-   * that contain bindable elements. This function is not idempotent.
+   * For each node in an array, scans it and its descendants for bindings.
+   * This function is not idempotent.
    *
    * Returns a promise that resolves after bindings have been added.
    *
-   * @param {!Node} node
-   * @return {!Promise}
+   * @param {!Array<!Node>} nodes
+   * @return {!Promise<number>}
    * @private
    */
-  addBindingsForNode_(node) {
-    // Limit number of total bindings (unless in local manual testing).
-    const limit = (getMode().localDev && !getMode().test)
-        ? Number.POSITIVE_INFINITY
-        : this.maxNumberOfBindings_ - this.numberOfBindings();
-    return this.scanNode_(node, limit).then(results => {
-      const {
-        boundElements, bindings, expressionToElements, limitExceeded,
-      } = results;
-
-      this.boundElements_ = this.boundElements_.concat(boundElements);
-      Object.assign(this.expressionToElements_, expressionToElements);
-
-      dev().fine(TAG, `Scanned ${bindings.length} bindings from ` +
-          `${boundElements.length} elements.`);
-
-      if (limitExceeded) {
-        user().error(TAG, 'Maximum number of bindings reached ' +
-            `(${this.maxNumberOfBindings_}). Additional elements with ` +
-            'bindings will be ignored.');
-      }
+  addBindingsForNodes_(nodes) {
+    // For each node, scan it for bindings and store them.
+    const scanPromises = nodes.map(node => {
+      // Limit number of total bindings (unless in local manual testing).
+      const limit = (getMode().localDev && !getMode().test)
+          ? Number.POSITIVE_INFINITY
+          : this.maxNumberOfBindings_ - this.numberOfBindings();
+
+      return this.scanNode_(node, limit).then(results => {
+        const {
+          boundElements, bindings, expressionToElements, limitExceeded,
+        } = results;
+
+        this.boundElements_ = this.boundElements_.concat(boundElements);
+        Object.assign(this.expressionToElements_, expressionToElements);
+
+        dev().fine(TAG, `Scanned ${bindings.length} bindings from ` +
+            `${boundElements.length} elements.`);
+
+        if (limitExceeded) {
+          user().error(TAG, 'Maximum number of bindings reached ' +
+              `(${this.maxNumberOfBindings_}). Additional elements with ` +
+              'bindings will be ignored.');
+        }
+
+        return bindings;
+      });
+    });
 
+    // Once all scans are complete, combine the bindings and ask web-worker to
+    // evaluate expressions in a single RPC.
+    return Promise.all(scanPromises).then(results => {
+      // `results[i]` is an array of bindings from the i'th scan -- combine.
+      const bindings = Array.prototype.concat.apply([], results);
```

Comment:
Results is just an array, so this can be `[].concat(results)`. But why duplicate?